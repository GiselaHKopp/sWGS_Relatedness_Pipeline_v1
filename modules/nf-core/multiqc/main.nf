process MULTIQC {
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/8c/8c6c120d559d7ee04c7442b61ad7cf5a9e8970be5feefb37d68eeaa60c1034eb/data' :
        'community.wave.seqera.io/library/multiqc:1.32--d58f60e4deb769bf' }"

    input:
    path  multiqc_files, stageAs: "?/*"
    path(multiqc_config)
    path(extra_multiqc_config)
    path(multiqc_logo)
    path(replace_names)
    path(sample_names)

    output:
    path "*multiqc_report.html", emit: report
    path "*_data"              , emit: data
    path "*_plots"             , optional:true, emit: plots
    path "versions.yml"        , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ? "--filename ${task.ext.prefix}.html" : ''
    def config = multiqc_config ? "--config $multiqc_config" : ''
    def extra_config = extra_multiqc_config ? "--config $extra_multiqc_config" : ''
    def logo = multiqc_logo ? "--cl-config 'custom_logo: \"${multiqc_logo}\"'" : ''
    def replace = replace_names ? "--replace-names ${replace_names}" : ''
    def samples = sample_names ? "--sample-names ${sample_names}" : ''
    """
    multiqc \\
        --force \\
        $args \\
        $config \\
        $prefix \\
        $extra_config \\
        $logo \\
        $replace \\
        $samples \\
        .

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        multiqc: \$( multiqc --version | sed -e "s/multiqc, version //g" )
    END_VERSIONS
    """

    stub:
    """
    mkdir multiqc_data
    mkdir multiqc_plots
    touch multiqc_report.html

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        multiqc: \$( multiqc --version | sed -e "s/multiqc, version //g" )
    END_VERSIONS
    """
}
