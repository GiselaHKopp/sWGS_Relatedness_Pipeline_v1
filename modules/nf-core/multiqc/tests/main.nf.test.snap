{
    "multiqc_versions_single": {
        "content": [
            [
                "versions.yml:md5,737bb2c7cad54ffc2ec020791dc48b8f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-10-27T13:33:24.356715"
    },
    "multiqc_stub": {
        "content": [
            [
                "multiqc_report.html",
                "multiqc_data",
                "multiqc_plots",
                "versions.yml:md5,737bb2c7cad54ffc2ec020791dc48b8f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-10-27T13:34:11.103619"
    },
    "multiqc_versions_config": {
        "content": [
            [
                "versions.yml:md5,737bb2c7cad54ffc2ec020791dc48b8f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-10-27T13:34:04.615233"
    }
}